battlepass/src/lib.rs

```diff
@@ -364,33 +365,40 @@ pub mod pallet {
 		/// - `price`: Price for the Battlepass subscription.
 		#[pallet::call_index(1)]
 		#[pallet::weight(<T as pallet::Config>::WeightInfo::update_battlepass())]
+		#[transactional]
 		pub fn update_battlepass(
 			origin: OriginFor<T>,
 			battlepass_id: T::Hash,
 			name: Option<String<T>>,
 			cid: Option<String<T>>,
 			price: Option<u16>,
 		) -> DispatchResult {
-			let creator = ensure_signed(origin)?;
+			let creator = ensure_signed(origin.clone())?;
 			// check if Battlepass exists
 			let mut battlepass = Self::get_battlepass(battlepass_id).ok_or(Error::<T>::BattlepassUnknown)?;
 			// check if there is something to update
+			let name_changed = name.is_some() && name.clone().unwrap() != battlepass.name;
+			let cid_changed = cid.is_some() && cid.clone().unwrap() != battlepass.cid;
+			let price_changed = price.is_some() && price.clone().unwrap() != battlepass.price;
 			ensure!(
-				name.is_some() && name.clone().unwrap() != battlepass.name || 
-				cid.is_some() && cid.clone().unwrap() != battlepass.cid || 
-				price.is_some() && price.clone().unwrap() != battlepass.price, 
+				name_changed || cid_changed || price_changed, 
 				Error::<T>::NoChangesProvided
 			);
 			// check if Battlepass state is not ENDED
 			ensure!(!Self::check_battlepass_state(battlepass_id, BattlepassState::ENDED)?, Error::<T>::BattlepassStateWrong);
 			// check if Org is active
 			ensure!(T::Control::is_org_active(&battlepass.org_id), Error::<T>::OrgUnknownOrInactive);
 			// check permissions (prime)
-			ensure!(Self::is_prime(&battlepass.org_id, creator.clone())?, Error::<T>::AuthorizationError);
+			ensure!(Self::is_prime_or_bot(&battlepass.org_id, creator.clone())?, Error::<T>::AuthorizationError);
 
-			if name.is_some() { battlepass.name = name.clone().unwrap() };
-			if cid.is_some() { battlepass.cid = cid.clone().unwrap() };
-			if price.is_some() { battlepass.price = price.clone().unwrap() };
+			let prime = T::Control::org_prime_account(&battlepass.org_id).ok_or(Error::<T>::OrgPrimeUnknown)?;
```

Comment:
`org_prime_account` storage is read multiple times during extrinsic execution which adds up uneffective weight to the call. I suggest to reuse this value in `is_prime_or_bot` or at least perform duplicated storage read only in case cid changes, since it is the only condition when this value is used.

Reply:
Good point, moved this line inside `if cid_changed` statement.